packages/default-theme/components/SwLogin.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div class="sw-login">
+    <div class="form sw-login__form">
+      <h2 class="sw-login__header">Log in</h2>
+      <SfAlert
+        v-if="error"
+        class="sw-login__alert"
+        type="danger"
+        :message="error"
+      />
+      <SfInput
+        v-model="login"
+        name="login"
+        label="Your login"
+        class="form__input"
+        :valid="!$v.login.$error"
+        :disabled="isLoading"
+        error-message="Login is required"
+        @blur="$v.login.$touch()"
+      />
+      <SfInput
+        v-model="password"
+        name="password"
+        label="Password"
+        type="password"
+        class="form__input"
+        :valid="!$v.password.$error"
+        :disabled="isLoading"
+        error-message="Password is required"
+        @blur="$v.password.$touch()"
+      />
+      <SfButton
+        class="sf-button--full-width form__button"
+        :disabled="isLoading"
+        @click="invokeLogin"
+      >
+        Login
+      </SfButton>
+    </div>
+  </div>
+</template>
+
+<script>
+import { SfInput, SfButton, SfAlert } from '@storefront-ui/vue'
+import { validationMixin } from 'vuelidate'
+import { required } from 'vuelidate/lib/validators'
+import { useUser } from '@shopware-pwa/composables'
+
+export default {
+  name: 'SwResetPassword',
+  components: { SfButton, SfInput, SfAlert },
+  mixins: [validationMixin],
+  data() {
+    return {
+      login: '',
+      password: ''
+    }
+  },
+  setup() {
+    const { login, loading, error } = useUser()
+    return {
+      clientLogin: login,
+      isLoading: loading,
+      error
+    }
+  },
+  validations: {
+    login: {
+      required
+    },
+    password: {
+      required
+    }
+  },
+  methods: {
+    async invokeLogin() {
+      this.$v.$touch()
+      if (this.$v.$invalid) {
+        this.error = 'Form is not valid'
```

Comment:
not user-friendly message
```suggestion
        this.error = 'Please provide your login and password.'
```

Reply:
and it should not be set here, but in alert and visible only when $v.$error
setting this here may cause problems with a correct error message and not cleaning it properly as most of the logic is in composable

Reply:
Ok, It was temporary solution. Currently after clicking register/login without filling any fields it will show errors individually for each field, so you right that code should be deleted.